DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/GroupDelegatesController.cs

```diff
@@ -0,0 +1,189 @@
+
+            var model = new AddGroupDelegateViewModel(
+                delegateUsers,
+                jobGroups,
+                customPrompts,
+                page,
+                groupId,
+                groupName!,
+                searchString,
+                filterBy
+            );
+
+            Response.UpdateOrDeleteFilterCookie(AddGroupDelegateCookieName, filterBy);
+            return View(model);
+        }
+
+        [HttpPost("Add/{delegateId:int}")]
+        [ServiceFilter(typeof(VerifyAdminUserCanAccessDelegateUser))]
+        public IActionResult AddDelegate(int groupId, int delegateId)
+        {
+            var delegateUser = userService.GetDelegateUserById(delegateId);
+            var adminId = User.GetAdminId();
+
+            groupsService.AddDelegateToGroupAndEnrolOnGroupCourses(
+                groupId,
+                delegateUser!,
+                adminId
+            );
+
+            return RedirectToAction("ConfirmDelegateAdded", new { groupId, delegateId });
+        }
+
+        [HttpGet("Add/{delegateId:int}/Confirmation")]
+        public IActionResult ConfirmDelegateAdded(int groupId, int delegateId)
+        {
+            var delegateUser = userService.GetDelegateUserById(delegateId);
+            if (delegateUser == null)
+            {
+                return NotFound();
+            }
+
+            var centreId = User.GetCentreId();
+            var groupName = groupsService.GetGroupName(groupId, centreId);
+
+            var model = new ConfirmDelegateAddedViewModel(delegateUser.FullName, groupName!, groupId);
+            return View(model);
+        }
+
+        [HttpGet("Add/SelectDelegate/AllItems")]
+        public IActionResult SelectDelegateAllItems(int groupId)
+        {
+            var centreId = User.GetCentreId();
+            var jobGroups = jobGroupsService.GetJobGroupsAlphabetical();
+            var customPrompts = centreCustomPromptHelper.GetCustomPromptsForCentre(centreId);
+            var delegateUsers = userService.GetDelegatesNotRegisteredForGroupByGroupId(groupId, centreId);
+
+            var model = new SelectDelegateAllItemsViewModel(
+                delegateUsers,
+                jobGroups,
+                customPrompts,
+                groupId
+            );
+
+            return View(model);
+        }
+
+
+        [HttpGet]
+        [Route("{delegateId:int}/Remove")]
+        public IActionResult RemoveGroupDelegate(int groupId, int delegateId)
+        {
+            var centreId = User.GetCentreId();
+            var groupName = groupsService.GetGroupName(groupId, centreId);
+            var groupDelegates = groupsService.GetGroupDelegates(groupId).ToList();
+            var delegateUser = groupDelegates.SingleOrDefault(gd => gd.DelegateId == delegateId);
+
+            if (groupName == null || delegateUser == null)
+            {
+                return NotFound();
+            }

```

Comment:
VerifyAdminUserCanAccessDelegateUser and remove this null check, once we have both filters, it becomes unnecessary